Source/Core/PolygonGeometry.js

```diff
@@ -372,18 +376,84 @@ define([
         return geometry;
     }
 
-    function computeRectangle(positions, ellipsoid, result) {
+    var startCartographicScratch = new Cartographic();
+    var endCartographicScratch = new Cartographic();
+    var idlCross = {
+        west : 0.0,
+        east : 0.0
+    };
+    function computeRectangle(positions, ellipsoid, arcType, granularity, result) {
+        result = defaultValue(result, new Rectangle());
         if (!defined(positions) || positions.length < 3) {
-            if (!defined(result)) {
-                return new Rectangle();
-            }
             result.west = 0.0;
             result.north = 0.0;
             result.south = 0.0;
             result.east = 0.0;
             return result;
         }
-        return Rectangle.fromCartesianArray(positions, ellipsoid, result);
+
+        result.west = Number.POSITIVE_INFINITY;
+        result.east = Number.NEGATIVE_INFINITY;
+        result.south = Number.POSITIVE_INFINITY;
+        result.north = Number.NEGATIVE_INFINITY;
+
+        idlCross.west = Number.POSITIVE_INFINITY;
+        idlCross.east = Number.NEGATIVE_INFINITY;
+
+        var inverseChordLength = 1.0 / CesiumMath.chordLength(granularity, ellipsoid.maximumRadius);
+        var positionsLength = positions.length;
+        var endCartographic = ellipsoid.cartesianToCartographic(positions[0], endCartographicScratch);
+        var startCartographic = startCartographicScratch;
+        var swap;
+
+        for (var i = 1; i < positionsLength; i++) {
+            swap = startCartographic;
+            startCartographic = endCartographic;
+            endCartographic = ellipsoid.cartesianToCartographic(positions[i], swap);
+            interpolateAndGrowRectangle(startCartographic, endCartographic, ellipsoid, arcType, inverseChordLength, result, idlCross);
+        }
+
+        swap = startCartographic;
+        startCartographic = endCartographic;
+        endCartographic = ellipsoid.cartesianToCartographic(positions[0], swap);
+        interpolateAndGrowRectangle(startCartographic, endCartographic, ellipsoid, arcType, inverseChordLength, result, idlCross);
+
+        if (result.east - result.west > idlCross.west - idlCross.east) {
+            result.east = idlCross.east;
+            result.west = idlCross.west;
+        }
+
+        return result;
+    }
+
+    var interpolatedCartographicScratch = new Cartographic();
+    function interpolateAndGrowRectangle(startCartographic, endCartographic, ellipsoid, arcType, inverseChordLength, result, idlCross) {
+        var segmentPath;
+        if (arcType === ArcType.GEODESIC) {
+            segmentPath = new EllipsoidGeodesic(startCartographic, endCartographic, ellipsoid);
+        } else {
+            segmentPath = new EllipsoidRhumbLine(startCartographic, endCartographic, ellipsoid);
```

Comment:
If it's using Rhumb lines, can't we use the previous logic of `return Rectangle.fromCartesianArray(positions, ellipsoid, result);` instead of iterating through the entire arc?